planet/src/main/webapp/js/itinerary.js

```diff
@@ -21,47 +49,133 @@ function renderStartingLocation() {
     }
 }
 
-async function renderEvents() {
-    const eventsResponse = await fetch('/update-event');
-    const events = await eventsResponse.json();
+// Add an event to the firebase realtime database
+// Todo: input validation, success/failure callbacks
+async function addEvent() {
+    const eventName = document.getElementById('add-event-name').value;
+    const eventAddress = document.getElementById('add-event-address').value;
+    const eventDuration = document.getElementById('add-event-duration').value;
+    const userId = firebase.auth().currentUser.uid; 
+    
+    // Set every event's opening hours to 8am - 5pm for now
+    const openingTime = TimeRange.getTimeInMinutes(8,0);
+    const closingTime = TimeRange.getTimeInMinutes(17,0);
+
+    const eventListRef = database.ref('users/' + userId + '/currentList');
+    // Get the order number by counting existing events
+    const snap = await eventListRef.once('value');
+    const order = snap.numChildren() + 1;
+
+    //Create a new event
+    const newEventRef = eventListRef.push();
+    newEventRef.set({
```

Comment:
does this persist the new event in the database?

Reply:
The push() method generates a unique reference key for the new event, and calling set() will overwrite the values in the new event and save it.

Reply:
SGTM